pkg/auth/dependency/mfa/pq/store.go

```diff
@@ -322,6 +322,45 @@ func (s *storeImpl) CreateBearerToken(a *mfa.BearerTokenAuthenticator) error {
 	return nil
 }
 
+func (s *storeImpl) GetBearerTokenByToken(userID string, token string) (*mfa.BearerTokenAuthenticator, error) {
+	q1 := s.sqlBuilder.Tenant().
+		Select(
+			"a.id",
+			"a.user_id",
+			"a.type",
+			"abt.parent_id",
+			"abt.token",
+			"abt.created_at",
+			"abt.expire_at",
+		).
+		From(s.sqlBuilder.FullTableName("authenticator"), "a").
+		Join(
+			s.sqlBuilder.FullTableName("authenticator_bearer_token"),
+			"abt",
+			"a.id = abt.id",
+		).
+		Where("a.user_id = ? AND abt.token = ?", userID, token)
```

Comment:
I think we need to compare the token in constant time.

Reply:
It requires us to fetch all tokens from database. The number can be unbound.

Reply:
I see, we do not limit amount of bearer tokens associated with a single authenticator. I think we should add a note here stating the limitation, since it may be a security risk.

Reply:
Sure.